File: java-app/src/main/java/com/showcase/java/domain/model/TaskStatus.java
package com.showcase.java.domain.model;

import java.time.Instant;

public sealed interface TaskStatus {
    record Open() implements TaskStatus {}
    record InProgress(UserId assignee) implements TaskStatus {}
    record Blocked(String reason) implements TaskStatus {}
    record Completed(Instant completedAt) implements TaskStatus {}
}

File: kotlin-app/src/main/kotlin/com/showcase/kotlin/domain/model/TaskStatus.kt
package com.showcase.kotlin.domain.model

import java.time.Instant

sealed class TaskStatus {
    data object Open : TaskStatus()
    data class InProgress(val assignee: UserId) : TaskStatus()
    data class Blocked(val reason: String) : TaskStatus()
    data class Completed(val completedAt: Instant) : TaskStatus()
}
